src/api/Feed.js

```diff
@@ -1128,6 +1261,45 @@ class Feed extends Base {
         }
     };
 
+    /**
+     * Deletes a reply from the cache
+     *
+     * @param {string} id - The id of the feed item to be deleted
+     * @param {string} parentId - The id of the parent feed item
+     * @param {Function} successCallback - function to be called after the delete
+     */
+    deleteReplyItem = (id: string, parentId: string, successCallback: Function = noop) => {
+        const cachedItems = this.getCachedItems(this.file.id);
+
+        const getItemWithFilteredReplies = <T: { replies?: Array<Comment> }>(item: T, replyId: string): T => {
+            if (!item.replies) {
+                return item;
+            }
+            const replies = item.replies.filter(reply => reply.id !== replyId);
+            return { ...item, replies };
+        };
+
+        if (cachedItems) {
+            const feedItems = cachedItems.items.map(item => {
+                if (item.id !== parentId) {
+                    return item;
+                }
+                if (item.type === 'annotation') {
+                    return getItemWithFilteredReplies<Annotation>(item, id);
```

Comment:
This looks similar to several other methods/helpers in this class. How much logic is left remaining after this change to complete our feature set? I think a refactor may be in order because this file is getting pretty enormous.

Reply:
I agree, `Feed` got pretty big and future refactoring is a good idea. Actually this PR is the last set of new methods and I don't anticipate adding more. Some changes and fixes might be needed when finalizing and testing FE, BFF and Domain Service takes place very shortly, but we shouldn't add anything new.